Script to verify agent OS, downloaded from Github release

chidozieononiwu opened the pull request:
Getting script from Github release rather than blob storage. This is a temporary solution till we can get it to work from the devops feed.

chidozieononiwu committed:
Script to verify agent OS, downloaded from github release

Script to verify agent OS, downloaded from github release

weshaggard commented on .azure-pipelines/client.yml:
```diff
@@ -87,6 +87,15 @@ jobs:
       vmImage: '$(OSVmImage)'
 
     steps:
+      - powershell:  'Invoke-WebRequest -Uri "$(VerifyAgentOSBlobPath)" -OutFile "./verify_agent_os.py"'
```
Where is VerifyAgentOSBlobPath defined?

I'm kind of expecting you would be downloading a general zip containing a set of tools that we download and extract into a tools directory and then the next set of steps would just reference stuff from that tools directory.

chidozieononiwu replied:
The URL is set as a Build Pipeline Variable. I see what you mean. So download all the sdk tools as a zip then extract that into a directory on the pipeline. I thought you didn't want us pulling in more tools than needed for the current job?

weshaggard replied:
>I thought you didn't want us pulling in more tools than needed for the current job?

In general I want to keep it minimal so we have to be careful of what we put into this set of tools but we need to balance the size of this toolset vs the complexity of manually downloading each tool/script individually. For now my goal is to have a single zip with a set of scripts we will be using in most of our definitions so that they can be bundled. Even if we don't use all the scripts in the bundle in a given build definition that should be OK but we should definitely be careful to not make the size too much.

weshaggard replied:
>The URL is set as a Build Pipeline Variable

I think we want the url to be in the yml file as part of the thing we need to ensure is that the exact version of the tool set is committed in the repo so we can reproduce the exact build if we need to go back and rebuild a older commit in the repo, which may be the case if we need to go back and hotfix and older build.

chidozieononiwu committed:
parent 1a1fc747bc7a292a3dd015b6a5082f692ccb81d3
author Chidozie Ononiwu <chononiw@microsoft.com> 1553691449 -0700
committer Chidozie Ononiwu <chononiw@microsoft.com> 1553705080 -0700

Downloading sdk-tools as archive and extracting it

Checking contents of directory to confirm download

Updated script working directory

Removed PS commands from quotes

Fixed Error in Script

Removed script to check directory content, adjusted scriptto two lines

Setting Pipeline to runin public

weshaggard commented on .azure-pipelines/client.yml:
```diff
@@ -87,6 +87,18 @@ jobs:
 
     steps:
+      - powershell: |
+          Invoke-WebRequest -Uri "https://github.com/chidozieononiwu/azure-sdk-for-net/releases/download/verifyAgentOS_v1.1/verify_agent_os.zip" `
```
Yes this is along the lines of what I was looking for. I'd like to get this pulling from the azure-sdk-tools release before we try to merge this however. I'm fine if you want to manually create a release there to bootstrap but ideally you would go ahead and make the necessary changes to azure-sdk-tools build pipeline to produce the release.

chidozieononiwu replied:
Ok, I will see if i can get the changes setup in the azure-sdk-tools today otherwise I will create it manually so that this PR can be merged. I think I also need to set this up for all the repos?

weshaggard replied:
Yes you will need to set it up for all repos which is why I'd like the url to closely match what we expect it to look like from the azure-sdk-tools build before updating all the repos.

weshaggard commented on .azure-pipelines/client.yml:
```diff
@@ -87,6 +87,18 @@ jobs:
     steps:
+      - powershell: |
+          Invoke-WebRequest -Uri "https://github.com/chidozieononiwu/azure-sdk-for-net/releases/download/verifyAgentOS_v1.1/verify_agent_os.zip" `
+          -OutFile "verify_agent_os.zip" | Wait-Process; Expand-Archive -Path "verify_agent_os.zip" -DestinationPath "./azure-sdk-tools/"
```
Is Wait-Process actually needed? I'd expect Invoke-WebRequest would block until it was done.

chidozieononiwu replied:
I though so too, but it skips along to try and extract the zip before the download completes.

weshaggard replied:
Did you find a reference online somewhere that prompted you to call wait-process? If so could you share the link?

weshaggard commented on .azure-pipelines/client.yml:
```diff
@@ -87,6 +87,18 @@ jobs:
+      - powershell: |
+          Invoke-WebRequest -Uri "https://github.com/chidozieononiwu/azure-sdk-for-net/releases/download/verifyAgentOS_v1.1/verify_agent_os.zip" `
+          -OutFile "verify_agent_os.zip" | Wait-Process; Expand-Archive -Path "verify_agent_os.zip" -DestinationPath "./azure-sdk-tools/"
+        workingDirectory: '$(System.DefaultWorkingDirectory)'
```
We should be careful about putting our tools in here as this is the same place that the source code is cloned into and we don't want the contents of our tools to interfere with that. I suggest using `Build.BinariesDirectory`, see https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#build-variables for more information.

chidozieononiwu replied:
Ok, I will update that.

chidozieononiwu committed:
Updated build pipeline to download tools from tools repo, and test on playgound

chidozieononiwu committed:
Updating build pipeline to run on public

chidozieononiwu committed:
Fixed typo in proj file
